echidna/scripts/tutorials/getting_started.py

```diff
@@ -0,0 +1,384 @@
+# over multiple parameters, it is best to construct a dictionary of `_low` and
+# `_high` values for each parameter and then pass this to the shrink method.
+
+# In[18]:
+
+shrink_dict = {"energy_mc_low": mu - 5.*sigma,
+               "energy_mc_high": mu + 5.*sigma,
+               "radial_mc_low": 0.0,
+               "radial_mc_high": 3500}
+spectrum_1.shrink(**shrink_dict)
+
+
+# Using the `sum` method, you can check to see how many events were cut.
+
+# In[19]:
+
+print spectrum_1.sum()
+
+
+# Import the smear class:
+
+# In[20]:
+
+import echidna.core.smear as smear
+
+
+# and create the smearer object.
+
+# In[21]:
+
+smearer = smear.EnergySmearRes()
+
+
+# By default the "weighted smear" method considers all bins within a
+# $\pm 5\sigma$ range. For the sake of speed, we will reduce this to 3 here.
+# Also set the energy resolution - 0.05 for 5%.
+
+# In[22]:
+
+smearer.set_num_sigma(3)
+smearer.set_resolution(0.05)
+
+
+# To smear our original spectrum and create the new `Spectra` object
+# `smeared_spectrum`:
+
+# In[23]:
+
+smeared_spectrum = smearer.weighted_smear(spectrum_1)
+
+
+# this should hopefully only create a couple of seconds.
+
+# The following code shows you how to make a simple script, using matplotlib,
+# to overlay the original and smeared spectra.
+
+# In[24]:
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def overlay_spectra(original, smeared, dimension="energy_mc", fignum=1):
+    """ Overlay original and smeared spectra.
+
+    Args:
+      original (echidna.core.spectra.Spectra): Original spectrum.
+      smeared (echidna.core.spectra.Spectra): Smeared spectrum.
+      dimension (string, optional): Dimension to project onto.
+        Default is "energy_mc".
+      fignum (int, optional): Figure number, if producing multiple
+        figures. Default is 1.
+
+    Returns:
+      matplotlib.figure.Figure: Figure showing overlaid spectra.
+    """
+    fig = plt.figure(num=fignum)
+    ax = fig.add_subplot(1, 1, 1)
+
+    par = original.get_config().get_par(dimension)
+    width = par.get_width()
+
+    # Define array of bin boundaries (1 more than number of bins)
+    bins = np.linspace(par._low, par._high, par._bins+1)
+    # Define array of bin centres
+    x = bins[:-1] + 0.5*width
+
+    # Overlay two spectra using projection as weight
+    ax.hist(x, bins, weights=original.project(dimension),
+            histtype="stepfilled", color="RoyalBlue",
+            alpha=0.5, label=original._name)
+    ax.hist(x, bins, weights=smeared.project(dimension), histtype="stepfilled",
+            color="Red", alpha=0.5, label=smeared._name)
+
+    # Add label/style
+    plt.legend(loc="upper right")
+    plt.ylim(ymin=0.0)
+    plt.xlabel(dimension + " [" + par.get_unit() + "]")
+    plt.ylabel("Events per " + str(width) + " " + par.get_unit() + " bin")
+    return fig
+
+
+# In[25]:
+
+fig1 = overlay_spectra(spectrum_1, smeared_spectrum, fignum=4)
+plt.show()
```

Comment:
The plot didn't appear. It was written:
ValueError                                Traceback (most recent call last)
<ipython-input-14-3378db744f5a> in <module>()
----> 1plot.plot_surface(spectrum_1, "energy_mc", "radial_mc")

/.data/snoplus/evelina/virtual_echidna/echidna/echidna/output/plot.pyc in plot_surface(spectra, dimension1, dimension2)
    192     """
    193     fig = plt.figure()
--> 194axis = fig.add_subplot(111, projection='3d')
    195     index1 = spectra.get_config().get_index(dimension1)
    196     index2 = spectra.get_config().get_index(dimension2)

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/figure.pyc in add_subplot(self, *args, **kwargs)
    944         else:
    945             projection_class, kwargs, key = process_projection_requirements(
--> 946                 self, *args, **kwargs)
    947 
    948             # try to find the axes with this key in the stack

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in process_projection_requirements(figure, *args, **kwargs)
     96 
     97     if isinstance(projection, six.string_types) or projection is None:
---> 98projection_class = get_projection_class(projection)
     99     elif hasattr(projection, '_as_mpl_axes'):
    100         projection_class, extra_kwargs = projection._as_mpl_axes()

/.data/snoplus/evelina/virtual_echidna/lib/python2.7/site-packages/matplotlib/projections/__init__.pyc in get_projection_class(projection)
     65         return projection_registry.get_projection_class(projection)
     66     except KeyError:
---> 67         raise ValueError("Unknown projection '%s'" % projection)
     68 
     69 

ValueError: Unknown projection '3d'